keras/preprocessing/image.py

```diff
@@ -933,11 +969,23 @@ def _recursive_list(subpath):
                             is_valid = True
                             break
                     if is_valid:
-                        self.classes[i] = self.class_indices[subdir]
-                        i += 1
-                        # add filename relative to directory
+                        if do_train_validation_split:
+                            # perform variant assignment and determine whether to ignore files from list
+                            hash_pct = _calc_hash_pct(fname)
+                            valid_validation = hash_pct < validation_pct and not is_training
```

Comment:
Instead of hashing, why not just generate random numbers? Do we really care about the randomness of the valid set?